packages/contracts-bedrock/deploy-config/devnetL1.json

```diff
@@ -8,7 +8,7 @@
   "p2pSequencerAddress": "0x9965507D1a55bcC2695C58ba16FB37d819B0A4dc",
   "batchInboxAddress": "0xff00000000000000000000000000000000000901",
   "batchSenderAddress": "0x3C44CdDdB6a900fa2b585dd299e03d12FA4293BC",
-  "batchInboxContractAddress": "0xff00000000000000000000000000000000000901",
+  "batchInboxContractAddress": "0x6C6c4A173D8A4f809c63c405e452F1560AeD9F61",
```

Comment:
Looks like this address does not appear when I run `cd op-e2e; go test -v ./... -run '^TestBatcher$'.` Am I doing something wrong?

Reply:
Can you share the log of running that test?

Reply:
Sure, here are the steps:

> make nuke
> cd op-e2e; go test -v ./... -run '^TestBatcher$' > test.log
> cat test.log  | grep 0x6C6c4A173D8A4f809c63c405e452F1560AeD9F61
> 

I would expected` test.log` to contain an occurrence of the address.

Reply:
Here are the logs.
[test.log](https://github.com/EspressoSystems/op-leader-election/files/12523470/test.log)

Reply:
It has a different address in your case. Not sure why. Have you run `make devnet-allocs` recently? I thought we would all get the same addresses but maybe not.

```
INFO [09-05|08:36:00.072] Setting account                          name=RoundRobinLeaderElection address=0xD0012DC379A2F26E3A533d715850523C50a8EC81
INFO [09-05|08:36:00.072] Setting storage                          name=RoundRobinLeaderElection key=0x0000000000000000000000000000000000000000000000000000000000000033 value=dead
INFO [09-05|08:36:00.072] Setting storage                          name=RoundRobinLeaderElection key=0x0000000000000000000000000000000000000000000000000000000000000065 value=03
INFO [09-05|08:36:00.072] Setting storage                          name=RoundRobinLeaderElection key=0x0000000000000000000000000000000000000000000000000000000000000000 value=02
INFO [09-05|08:36:00.073] Setting account                          name=RoundRobinLeaderElectionProxy     address=0xf91A5cf8A42e226F57986caf882e815794a09ED5
INFO [09-05|08:36:00.073] Setting storage                          name=RoundRobinLeaderElectionProxy     key=0x0000000000000000000000000000000000000000000000000000000000000033 value=a0ee7a142d267c1f36714e4a8f75612f20a79720
INFO [09-05|08:36:00.073] Setting storage                          name=RoundRobinLeaderElectionProxy     key=0x0000000000000000000000000000000000000000000000000000000000000065 value=04
INFO [09-05|08:36:00.073] Setting storage                          name=RoundRobinLeaderElectionProxy     key=0x0000000000000000000000000000000000000000000000000000000000000066 value=05
INFO [09-05|08:36:00.073] Setting storage                          name=RoundRobinLeaderElectionProxy     key=0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc value=d0012dc379a2f26e3a533d715850523c50a8ec81
INFO [09-05|08:36:00.073] Setting storage                          name=RoundRobinLeaderElectionProxy     key=0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103 value=f16330a2e8f30bc53c7b9511a74aa1d29e4bf6a5
INFO [09-05|08:36:00.073] Setting storage                          name=RoundRobinLeaderElectionProxy     key=0x0000000000000000000000000000000000000000000000000000000000000000 value=02
```

Reply:
```
INFO [09-05|08:36:00.072] Setting account                          name=RoundRobinLeaderElection address=0xD0012DC379A2F26E3A533d715850523C50a8EC81
INFO [09-05|08:36:00.072] Setting storage                          name=RoundRobinLeaderElection key=0x0000000000000000000000000000000000000000000000000000000000000033 value=dead
INFO [09-05|08:36:00.072] Setting storage                          name=RoundRobinLeaderElection key=0x0000000000000000000000000000000000000000000000000000000000000065 value=03
INFO [09-05|08:36:00.072] Setting storage                          name=RoundRobinLeaderElection key=0x0000000000000000000000000000000000000000000000000000000000000000 value=02
INFO [09-05|08:36:00.073] Setting account                          name=RoundRobinLeaderElectionProxy     address=0xf91A5cf8A42e226F57986caf882e815794a09ED5
INFO [09-05|08:36:00.073] Setting storage                          name=RoundRobinLeaderElectionProxy     key=0x0000000000000000000000000000000000000000000000000000000000000033 value=a0ee7a142d267c1f36714e4a8f75612f20a79720
INFO [09-05|08:36:00.073] Setting storage                          name=RoundRobinLeaderElectionProxy     key=0x0000000000000000000000000000000000000000000000000000000000000065 value=04
INFO [09-05|08:36:00.073] Setting storage                          name=RoundRobinLeaderElectionProxy     key=0x0000000000000000000000000000000000000000000000000000000000000066 value=05
INFO [09-05|08:36:00.073] Setting storage                          name=RoundRobinLeaderElectionProxy     key=0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc value=d0012dc379a2f26e3a533d715850523c50a8ec81
INFO [09-05|08:36:00.073] Setting storage                          name=RoundRobinLeaderElectionProxy     key=0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103 value=f16330a2e8f30bc53c7b9511a74aa1d29e4bf6a5
INFO [09-05|08:36:00.073] Setting storage                          name=RoundRobinLeaderElectionProxy     key=0x0000000000000000000000000000000000000000000000000000000000000000 value=02
INFO [09-05|08:36:00.073] Setting account                          
```

Reply:
It looks like I'm getting different addresses each time I run `make devnet-allocs`, so maybe this doesn't work.

Reply:
Is the chain being recreated from genesis each time? I wonder if there's a way to force operations to run in series until deployment is complete...

Reply:
A possible work around could be to deploy the leader election contract in the integration test (I am doing this right now to test the bindings) and feed the address of the contract to the SystemConfig.

Reply:
Hmm no this does not look like a good idea as the Leader Election contract is deployed and initialized in [Deploy.sol](https://github.com/EspressoSystems/op-leader-election/blob/op-leader-election/packages/contracts-bedrock/scripts/Deploy.s.sol).

Reply:
The test in 3ab563d5b7b13bb1e75fd76b312d55c3c6ff7b75 suggests that the right address for the Leader Election contract is available from the config object.

Reply:
I agree with that conclusion, but I don't really understand how that works, given the inconsistent addresses we observed. What is different between `make devnet-allocs` and the e2e setup?

Reply:
Set the address of the batch inbox contract back to 0xff00000000000000000000000000000000000901 in devnetL1.json for now (see commit [3d144c3](https://github.com/EspressoSystems/op-leader-election/pull/67/commits/3d144c398ab65e9f19a7988c04c9be5bb88a67e2)). Created  ticket #70 to see if there is a way to deploy this contract in  a deterministic manner.